### PR TITLE
Add pull request template

### LINKED ISSUE #5502: Add Pull Request Template
- Currently, there is only a bug report template.
- Add a PR template.

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXA-2008)
┆Issue Number: FXA-2008

### PR DESCRIPTION
**Pull request by imskr**
Closes #5502 

- Add Pull Request Template feature

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+- [ ] Documentation update (Documentation content changed)
+- [ ] Other (please describe):
+
+## Checklist
+
+_Put an `x` in the boxes that apply_
+
+- [ ] I have read the [CONTRIBUTING](https://github.com/mozilla/fxa/blob/master/CONTRIBUTING.md) guidelines
```
Comment (jodyheavener) [inline .github/PULL_REQUEST_TEMPLATE.md:28]: I feel as though this would get cumbersome with each PR. Also, if it's your first time opening a PR in the repo GitHub will prompt you to read the Contributing doc.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+- [ ] Lint and unit tests pass locally with my changes
```
Comment (jodyheavener) [inline .github/PULL_REQUEST_TEMPLATE.md:29]: We use lint-staged to lint files before they can be commited, and I believe we also fail CI if linting catches anything, so this might be redundant

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+- [ ] I have added necessary documentation (if appropriate)
+- [ ] My changes generate no new warnings
```
Comment (jodyheavener) [inline .github/PULL_REQUEST_TEMPLATE.md:31]: This is a little ambiguous -- is this browser console warnings, build warnings, something else?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (imskr) [reply .github/PULL_REQUEST_TEMPLATE.md:28]: Yeah, but I think it is useful if someone is returning contributor. But if it's not necessary I can remove it.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (imskr) [reply .github/PULL_REQUEST_TEMPLATE.md:29]: Cool, I will remove it.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (imskr) [reply .github/PULL_REQUEST_TEMPLATE.md:31]: It is for if there is some warning while building.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (LZoog) [reply .github/PULL_REQUEST_TEMPLATE.md:28]: I don't mind this checkbox in particular.

It could also be useful to have a checkbox for, "my commit is GPG signed." Generally that's where I see issues.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+- Basic tests done to validate
+- Conflicts if any (describe the reason for conflict)
+
+## Types of changes
+
+_Put an `x` in the boxes that apply_
+
+- [ ] Bugfix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
```
Comment (LZoog) [inline .github/PULL_REQUEST_TEMPLATE.md:20]: I don't know that contributors ever submit these types of changes. I would actually say we can remove "types of changes" section altogether - if they've read the contributor guidelines, they know to preface their commit with "type of change" such as `feat(content-server):` or `chore(docs)` which covers this.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
+## Issue that this pull request solves
+
+Closes: # (issue number)
+
+## Proposed changes
+
+Brief description of what is fixed or changed
+
+- Files changed
```
Comment (LZoog) [inline .github/PULL_REQUEST_TEMPLATE.md:9]: A contributor might think they need to list all paths here that changed, which can be super cumbersome (I used to be on a project that required this and I can say first hand, it's annoying). I like having "Proposed changes" and "Brief description of what is fixed or changed," but maybe we remove all the bullet points"?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (LZoog) [reply .github/PULL_REQUEST_TEMPLATE.md:29]: I agree we should take out the linting bit, but we could instead rephrase this to be something like, "If applicable, I have modified or added tests which pass locally."

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (LZoog) [reply .github/PULL_REQUEST_TEMPLATE.md:31]: I think generally, this would be covered with passing tests, except for when there's a new console warning in the browser. I agree it could be a little more specific, but I also don't want a contributor to be blocked on this point if they're not sure what this means or that their PR would be rejected if they don't check this box. Kind of indifferent here - I'm OK with "If applicable, my changes generate no new console warnings," or just removing this altogether. I don't typically run a build on the project before submitting a PR and wouldn't expect a contributor to run a build before their changes and compare warnings with after their changes.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+_Put an `x` in the boxes that apply_
+
+- [ ] I have read the [CONTRIBUTING](https://github.com/mozilla/fxa/blob/master/CONTRIBUTING.md) guidelines
+- [ ] Lint and unit tests pass locally with my changes
+- [ ] I have added necessary documentation (if appropriate)
+- [ ] My changes generate no new warnings
+
+## Screenshots
+
+Please attach the screenshots of the changes made in case of change in user interface
```
Comment (LZoog) [inline .github/PULL_REQUEST_TEMPLATE.md:35]: You could add a period here and on the last point. You could also note that this + the last point are optional.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (imskr) [reply .github/PULL_REQUEST_TEMPLATE.md:28]: Thanks I will add gpg and remove contributing one

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (imskr) [reply .github/PULL_REQUEST_TEMPLATE.md:31]: Noted. I will remove it.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 4 above)
Comment (imskr) [reply .github/PULL_REQUEST_TEMPLATE.md:20]: I will remove the section

**Comment by imskr** [comment]
@jodyheavener @LZoog I've made the changes as requested. There are other changes too due to prettier lint. Please review.

**Comment by vbudhram** [comment]
@imskr Looks like you committed some tailwind files? These should be removed.

**Comment by imskr** [comment]
@vbudhram It got committed automatically when running the lint. I will delete it.

**Comment by jodyheavener** [comment]
Huh, was it a `tailwind.out` file? This should be ignored in Git. If you encounter it again ping me.

**Comment by imskr** [comment]
@jodyheavener Yes it was `tailwind.out`. I deleted it. I will ping you if i encounter this again!
